controllers/awsadapterconfig_controller.go

```diff
@@ -84,20 +92,44 @@ func (r *AWSAdapterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return r.updateLastPollStatusFailure(ctx, objOld, "error occurred while loading aws sdk config", err, &l, time.Now())
 	}
 	l.Info("AWS SDK config loaded successfully")
-	svc := eks.NewFromConfig(cfg)
+
+	stsClient := sts.NewFromConfig(cfg)
+	ec2Client := ec2.NewFromConfig(cfg)
+	ecrClient := ecr.NewFromConfig(cfg)
+	eksClient := eks.NewFromConfig(cfg)
+	inspector2Client := inspector2.NewFromConfig(cfg)
 
 	objNew := objOld.DeepCopy()
 	objNew.Status.EKSCluster = &securityv1alpha1.EKSCluster{}
+	objNew.Status.AccountData = &securityv1alpha1.AccountData{}
 
 	clusterFound := false
 
-	if x, err := svc.ListClusters(context.TODO(), &eks.ListClustersInput{}); err == nil {
+	if callerIdentity, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{}); err == nil && callerIdentity.Account != nil {
```

Comment:
In general, it might be a better thing to reduce code nesting by using error != nil instead of err==nil construct. Especially if we are anyways returning prematurely on error at multiple places (which is yet another problem though). 

E.g:
if err != nil {
logerror and return
}
later part,